### PR TITLE
IOS-12035: Allow to download file object excluding specific file fields

### PR DESCRIPTION
**Pull request by thuyBox**
equivalent to https://github.com/box/box-ios-sdk/pull/245 on master

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @thuyBox, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by thuyBox** [comment]
test this please

**Comment by boxtcarpel** [comment]
+2 LGTM

🔵

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.m [hunk 1]
```diff
@@ -38,7 +38,7 @@ - (BOXAPIOperation *)createOperation
     NSMutableDictionary *parameters = [NSMutableDictionary dictionary];
     
     if (self.requestAllItemFields) {
-        parameters[BOXAPIParameterKeyFields] = [self fullCommentFieldsParameterString];
+        parameters[BOXAPIParameterKeyFields] = [self fullItemFieldsParameterStringExcludingFields:self.fieldsToExclude];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.m]: This doesn't seem right, before it was fullCommentFields, now it's itemFields. I don't think that's what we want

File: BoxContentSDK/BoxContentSDK/Requests/BOXCommentRequest.m [hunk 2]
```diff
@@ -34,9 +34,9 @@ - (BOXAPIOperation *)createOperation
                                  subID:nil];
     
     NSMutableDictionary *parameters = [NSMutableDictionary dictionary];
-    
+
     if (self.requestAllItemFields) {
-        parameters[BOXAPIParameterKeyFields] = [self fullCommentFieldsParameterString];
+        parameters[BOXAPIParameterKeyFields] = [self fullItemFieldsParameterStringExcludingFields:self.fieldsToExclude];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/Requests/BOXCommentRequest.m]: comment fields

**Comment by boxtcarpel** [comment]
+2 LGTM

Comments on comments

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.m (hunk 1 above)
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileCommentsRequest.m]: done

File: BoxContentSDK/BoxContentSDK/Requests/BOXCommentRequest.m (hunk 2 above)
Comment (thuyBox) [inline BoxContentSDK/BoxContentSDK/Requests/BOXCommentRequest.m]: done

**Comment by boxtcarpel** [comment]
+2 LGTM

💫

**Pull request merged**